src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -154,6 +161,28 @@ def create_privacy_request(
             privacy_request.cache_identity(privacy_request_data.identity)
             privacy_request.cache_encryption(privacy_request_data.encryption_key)
 
+            # Store masking secrets in the cache
+            logger.info(
+                f"Caching masking secrets for privacy request {privacy_request.id}"
+            )
+            erasure_rules: List[Rule] = policy.get_rules_for_action(
+                action_type=ActionType.erasure
+            )
+            unique_masking_strategies_by_name: Set[str] = set()
+            for rule in erasure_rules:
+                unique_masking_strategies_by_name.add(rule.masking_strategy["strategy"])
+            for strategy_name in unique_masking_strategies_by_name:
+                strategy = SupportedMaskingStrategies[strategy_name].value
```

Comment:
This will break if `strategy_name` is unsupported, and therefore not a key in `SupportedMaskingStrategies` — although I think that may already return an error at schema level instead?

Reply:
it would error at schema level, but for true zero trust, I can do a check here too.

Reply:
ended up using `get_strategy(strategy_name, {})` which will throw `NoSuchStrategyException` if strategy does not exist